### ISSUE TITLE
Recall and Accuracy produce different values when used with logits

### ISSUE THREAD

**Issue by asusdisciple** [Lightning-AI/torchmetrics#2655]
## 🐛 Bug
When I use 
`recall = torchmetrics.Recall(task="multiclass", num_classes=4, average="macro")` 
with raw logits the result is different to using 
`cls = nn.functional.softmax(logits, dim=0).argmax(dim=1)`
and then plugging `cls` as input for Recall.

So `recall(cls)` and `recall(logits)` produce different results. Is this to be expected?

### To Reproduce

See above

### Expected behavior
I would expect the same results for the logits and softmax values.

### Environment

- Torch 2.3.1
- Lightning  2.3.3
- Python 3.10
